### ISSUE TITLE
[ feature request ] schema

### ISSUE THREAD

**Issue by yonas** [ruuda/rcl#60]
RCL is awesome. I'd like to be able to specify a schema to validate the output of RCL.

I could output to JSON and write a separate [JSON Schema](https://json-schema.org). If RCL supported JSON Schema, I would be saved from using another tool.

If RCL has it's own schema language, that would work too.

**Comment by denfren** [comment]
Sounds the like planned [record types](https://docs.ruuda.nl/rcl/types/#record-types) feature will help with that.

Once you have that type you could do
```
let out: TYPE_HERE = { ... };
out
```

**Comment by ruuda** [comment]
Yep, this is exactly the idea behind the type system! It will not map 1:1 to JSON Schema, but you will be able to enforce similar things with it. The way I imagine it’s most useful is not just to validate that a value matches a schema all the way at the top level, but also to annotate intermediate values to catch errors early.

This will take some more work, I’ll keep this issue open until record types are implemented.
